### PR TITLE
[Storage] Fix issue#10302 to support guess content-type when synchronizing files

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**  

Fix issue #10302 
After upgrading to new azcopy version, it will support guessing content-type

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S173

File: src/azure-cli/azure/cli/command_modules/storage/azcopy/util.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 STORAGE_RESOURCE_ENDPOINT = "https://storage.azure.com"
 SERVICES = {'blob', 'file'}
-AZCOPY_VERSION = '10.3.3'
+AZCOPY_VERSION = '10.5.0'
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/storage/azcopy/util.py:26]: Do we have test to cover this change?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/azcopy/util.py:26]: yes. we have live test only.

File: src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_azcopy_scenarios.py [hunk 2]
```diff
@@ -106,9 +115,6 @@ def test_storage_blob_azcopy_remove(self, resource_group, storage_account_info,
         self.cmd('storage remove -c {} -n readme --account-name {}'.format(
             container, storage_account))
 
-        self.cmd('storage remove -c {} -n readme --account-name {}'.format(
-            container, storage_account), expect_failure=True)
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_azcopy_scenarios.py:110]: still keep this command and remove "expect_failure"

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_azcopy_scenarios.py:110]: done

**Pull request merged**